azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodBuilder.java

```diff
@@ -0,0 +1,189 @@
+package azkaban.container.models;
+
+import io.kubernetes.client.custom.Quantity;
+import io.kubernetes.client.openapi.models.V1Container;
+import io.kubernetes.client.openapi.models.V1ContainerBuilder;
+import io.kubernetes.client.openapi.models.V1EnvVar;
+import io.kubernetes.client.openapi.models.V1EnvVarBuilder;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1ObjectMetaBuilder;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodBuilder;
+import io.kubernetes.client.openapi.models.V1ResourceRequirements;
+import io.kubernetes.client.openapi.models.V1ResourceRequirementsBuilder;
+import io.kubernetes.client.openapi.models.V1Volume;
+import io.kubernetes.client.openapi.models.V1VolumeBuilder;
+import io.kubernetes.client.openapi.models.V1VolumeMount;
+import io.kubernetes.client.openapi.models.V1VolumeMountBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A wrapper Kubernetes Pod builder which internally utilizes
+ * @see io.kubernetes.client.openapi.models to create kubernetes "Pod" resource
+ *
+ * This class provides custom builder methods to have tight coupling of
+ * volumes and init containers (associated with job-types) with application-container
+ * enabling better type safety and compact implementation
+ *
+ * The Pod-spec created using this builder creates one application-container
+ * and multiple init containers corresponding to each jobType
+ */
+public class AzKubernetesV1PodBuilder {
+    private static final String JOBTYPE_VOLUME_PREFIX = "jobtype-volume-";
+    private static final String JOBTYPE_INIT_PREFIX = "jobtype-init-";
+    private static final String AZ_CLUSTER_KEY = "AZ_CLUSTER";
+    private static final String AZ_CONF_VERSION_KEY = "AZ_CONF_VERSION";
+    private static final String JOBTYPE_MOUNT_PATH_KEY = "JOBTYPE_MOUNT_PATH";
+    private static final String DEFAULT_RESTART_POLICY = "Never";
+    private static final String API_VERSION = "v1";
+    private static final String KIND = "Pod";
```

Comment:
Can we please create Deployment type? It is preferable over POD.

Reply:
This kind field is not a configurable field. It will always be constant with respect to this builder class.

Reply:
Yes. Can you please change it to Deployment?